app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -229,13 +229,25 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
 
         //If the totalStar is positive, it implies that the skill has been rated
         //If so, set up the section to display the statistics else simply display a message for unrated skill
-        if (skillData.skillRating?.stars?.totalStar!! > 0) {
-            fiveStarAverageSkillRating = tv_average_rating
-            fiveStarTotalSkillRating = tv_total_rating
-
-            fiveStarTotalSkillRating.text = skillData.skillRating?.stars?.totalStar.toString()
-            fiveStarAverageSkillRating.text = skillData.skillRating?.stars?.averageStar.toString()
-            setSkillGraph()
+        if (skillData.skillRating != null) {
+            if (skillData.skillRating?.stars != null) {
+                if (skillData.skillRating?.stars?.totalStar!! > 0) {
```

Comment:
@iamareebjamal I did not check totalStar, rather the members of the stars object for null values because they have not been initialised to null. If the stars object is not null, in that case all the values inside it would bear the initial values. Here are the screenshots from the [FiveStarRateSkillService.java](https://github.com/fossasia/susi_server/blob/development/src/ai/susi/server/api/cms/FiveStarRateSkillService.java) file from the [server](https://github.com/fossasia/susi_server) : 

Here, the stars object has been initialised with null value.

<img width="1212" alt="screen shot 2018-06-24 at 1 52 16 pm" src="https://user-images.githubusercontent.com/30979369/41817324-59ec6df2-77b6-11e8-8f57-f5cd11d8d538.png">


Here, the members of the stars object have been initialised with zero values.

<img width="1220" alt="screen shot 2018-06-24 at 1 51 04 pm" src="https://user-images.githubusercontent.com/30979369/41817323-50883fc0-77b6-11e8-9cb5-2d6a31c74b9b.png">


This was why I did it this way. I asserted totalStar to non-null for this reason and hence, there shouldn't be a Null Pointer Exception as per my understanding.
In any case, if I am missing something and if there is some reason that we should check it for null values, please let me know, that would be very helpful.

Reply:
`totalStar!!` Either totalStar should be non-nullable or have a null check

Double exclamation in kotlin is shooting yourself in the foot

Reply:
Okay @iamareebjamal . Thanks, will change that asap. :)